frontend/public/components/storage/attach-storage.tsx

```diff
@@ -0,0 +1,437 @@
+import {
+  k8sGet,
+  k8sUpdate,
+  K8sResourceKind,
+  referenceFor,
+} from "../../module/k8s";
+import * as k8sModels from "../../models";
+import {
+  ButtonBar,
+  history,
+  resourceObjPath,
+  ListDropdown,
+  ResourceIcon
+} from "../utils";
+import { formatNamespacedRouteForResource } from "../../ui/ui-actions";
+import {
+  getMountPaths,
+  createVolumeMount,
+  createVolume,
+  getKindFromPlural
+} from "../storage/storage";
+
+const Checkbox = ({ value, checked, onChange }) => (
+  <div>
+    <label className="checkbox-label">
+      <input
+        type="checkbox"
+        onChange={({ target: { checked: newChecked } }) =>
+          onChange(value, newChecked)
+        }
+        checked={!!checked}
+      />
+      &nbsp;&nbsp;
+      {value}
+    </label>
+  </div>
+);
+
+export const PvcDropdown = props => {
+  const kind = "PersistentVolumeClaim";
+  const namespace = props.namespace;
+  const resources = [{ kind, namespace }];
+  const selectedKey = props.selectedKey;
+  return (
+    <ListDropdown
+      {...props}
+      desc="Storage"
+      resources={resources}
+      selectedKeyKind={kind}
+      placeholder="Add or create storage"
+      selectedKey={selectedKey}
+    />
+  );
+};
+
+const StorageForm = () =>
+  class StorageFormComponent extends React.Component<
+    StorageProps,
+    StorageState
+  > {
+    constructor(props) {
+      super(props);
+      const existingResource = _.pick(props.obj, ["metadata", "spec"]);
+      const resourceObj = _.defaultsDeep({}, props.fixed, existingResource, {
+        apiVersion: "v1",
+        data: {},
+        kind: props.kind,
+        metadata: {
+          name: ""
+        },
+        spec: {}
+      });
+
+      this.state = {
+        resourceObj: resourceObj,
+        inProgress: false,
+        stringData: _.mapValues(_.get(props.obj, "data"), value => {
+          return value ? window.atob(value) : "";
+        }),
+        disableForm: false
+      };
+      this.onVolumeNameChanged = this.onVolumeNameChanged.bind(this);
+      this.onMountPathChanged = this.onMountPathChanged.bind(this);
+      this.onSubPathChanged = this.onSubPathChanged.bind(this);
+      this.changePvcName = this.changePvcName.bind(this);
+      this.onReadOnlyChanged = this.onReadOnlyChanged.bind(this);
+      this.onError = this.onError.bind(this);
+      this.onFormDisable = this.onFormDisable.bind(this);
+      this.save = this.save.bind(this);
+    }
+    componentDidMount() {
+      const { kind, metadata } = this.state.resourceObj;
+      const ko = _.find(k8sModels, model => model.plural === kind);
+      k8sGet(ko, metadata.name, metadata.namespace).then(
+        obj => {
+          this.setState({ resourceObj: obj });
+        },
+        err => this.setState({ error: err.message, inProgress: false })
+      );
+    }
+    onError(err) {
+      this.setState({
+        error: err,
+        inProgress: false
+      });
+    }
+    changePvcName(event) {
```

Comment:
If you use this syntax, you can get rid of all the `bind()` calls in the constructor.

```ts
  changePvcName: React.ReactEventHandler<HTMLInputElement> = event => {
    // ...
  }
```